pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -402,12 +402,16 @@ func (s *Controller) serviceEntryHandler(_, curr config.Config, event model.Even
 		configsUpdated[makeConfigKey(svc)] = struct{}{}
 	}
 	// If service entry is deleted, call SvcUpdate to cleanup endpoint shards for services.
+	partiallyDeletedSvcs := []*model.Service{}
 	for _, svc := range deletedSvcs {
 		instanceKey := instancesKey{namespace: svc.Attributes.Namespace, hostname: svc.Hostname}
 		// There can be multiple service entries of same host reside in same namespace.
 		// Delete endpoint shards only if there are no service instances.
 		if len(s.serviceInstances.getByKey(instanceKey)) == 0 {
 			s.XdsUpdater.SvcUpdate(shard, string(svc.Hostname), svc.Attributes.Namespace, model.EventDelete)
+		} else {
+			// If there are some endpoints remaining for the host, trigger eds cache update
+			partiallyDeletedSvcs = append(partiallyDeletedSvcs, svc)
```

Comment:
nit: s/partiallyDeletedSvcs/updatedSvcs/g

Reply:
It's not a nit.. Appending to existing `updatedSvcs` will clean up a lot of new code here.